pkg/boot/boottest/json.go

```diff
@@ -21,17 +21,45 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+	"net/url"
 	"os"
+	"strings"
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/u-root/u-root/pkg/boot"
 	"github.com/u-root/u-root/pkg/curl"
 )
 
+// makeFileRel converts file:/// schemes to a path relative to the current
```

Comment:
Was this change absolutely required for the subject of the PR? I'm approving anyway but still ...

Reply:
Yes, the code takes a url.URL, converts it to a string, then parses it back to a url.URL. We were using a relative path in the url.URL previously which is non-standard, so Go adds an additional / in front which broke all the tests. Wikipedia says it best: "while "file:///foo.txt" is valid, "file://foo.txt" is not, although some interpreters manage to handle the latter".

Reply:
I split this out into another, dependent PR: https://github.com/u-root/u-root/pull/1865